server/punishments.ts

```diff
@@ -491,8 +491,8 @@ export const Punishments = new class {
 		return success;
 	}
 
-	roomPunish(room: Room | string, user: User, punishment: Punishment) {
-		const roomid = typeof room === 'string' ? room : room.id;
+	roomPunish(room: Room | RoomID, user: User, punishment: Punishment) {
+		const roomid = room instanceof Rooms.BasicRoom ? room.id : room;
```

Comment:
Please don't; we intentionally try to avoid `instanceof` so we can duck-type where necessary.

Reply:
(That goes for this and every other `typeof` you converted to `instanceof`.)